chat-plugins/mafia.js

```diff
@@ -1,919 +1,916 @@
-		}
-
-		return output + '</center></div>';
+	sendRoom(message, options = {}) {
+		if (options.uhtml) return this.room.add(`|uhtml|mafia|${message}`).update();
+		if (options.declare) return this.room.add(`|raw|<div class="broadcast-blue">${message}</div>`).update();
+		return this.room.add(message).update();
 	}
 
-	updatePregame() {
-		for (let i in this.room.users) {
-			let user = this.room.users[i];
-
-			if (user.userid in this.players) {
-				user.sendTo(this.room, '|uhtmlchange|mafia' + this.room.gameNumber + 'pregame|' + this.pregameWindow(true));
-			} else {
-				user.sendTo(this.room, '|uhtmlchange|mafia' + this.room.gameNumber + 'pregame|' + this.pregameWindow(false));
-			}
+	roomWindow() {
+		if (this.ended) return `<div class="infobox">The game of ${this.title} has ended.</div>`;
+		let output = `<div class="broadcast-blue">`;
+		if (this.phase === 'signups') {
+			output += `<h2 style="text-align: center">A game of ${this.title} was created</h2><p style="text-align: center"><button class="button" name="send" value="/mafia join">Join the game</button> <button class="button" name="send" value="/join view-mafia-${this.room.id}">Spectate the game</button></p>`;
+		} else {
+			output += `<p style="font-weight: bold">A game of ${this.title} is in progress. <button class="button" name="send" value="/mafia sub ${this.room.id}, in">Become a substitute</button> <button class="button" name="send" value="/join view-mafia-${this.room.id}">Spectate the game</button></p>`;
 		}
-	}
-
-	// UI
-
-	// Simple window, used for announcements and the likes.
-	mafiaWindow(image, content) {
-		let output = '<div class="broadcast-blue">';
-		output += '<h3>' + ((this.gamestate === 'night') ? 'Night ' : 'Day ') + this.day + '</h3>';
-		output += '<table><tr><td style="text-align:center;">' + image + '</td><td style="text-align:center;width:100%;">';
-		output += content;
-		output += '</td></tr></table></div>';
+		output += `</div>`;
 		return output;
 	}
 
-	announcementWindow(image, content) {
-		this.room.add('|html|' + this.mafiaWindow(image, content));
-		this.room.update();
-	}
-
-	displayGamestate() {
-		let alive = [];
-		let mafia = [];
-
-		for (let i in this.players) {
-			alive.push(this.players[i].name);
-			if (this.players[i].class.side === 'mafia') {
-				mafia.push(this.players[i].name);
-			}
-		}
-
-		let content = '<strong>Roles:</strong> ' + this.roleString + '<br /><strong>Alive:</strong> ' + Chat.escapeHTML(alive.join(', ')) + '<br /><strong>Dead:</strong> ' + Chat.escapeHTML(this.dead.join(', '));
-
-		for (let i in this.players) {
-			if (this.players[i].class.side === 'mafia') {
-				this.players[i].sendRoom('|html|' + this.mafiaWindow(this.players[i].class.image, content + '<br /><strong>Mafia:</strong> ' + Chat.escapeHTML(mafia.join(', '))));
-			} else {
-				this.players[i].sendRoom('|html|' + this.mafiaWindow(this.players[i].class.image, content));
-			}
-		}
+	onChatMessage(message, user) {
```

Comment:
I had no idea this was a thing, neat! (this is actually really helpful to know about)